src/background.js

```diff
@@ -1,29 +1,33 @@
 window.scrNoti = window.scrNoti || {};
+const seenMessages = {};
+
+// TODO:
+// onQuit
+// mark read?
+// Update seenMessages when changing settings: simple/extended and folders
 
 //==========================================
 // On new email...
 //==========================================
 window.scrNoti.newEmailListener = async (folder, messages) => {
-  let hasNotJunk = false;
 
   if (!(await window.scrNoti.isFolderToCheck(folder))) {
     return;
   }
 
+  // Find new messages, which have not been seen yet
   if (messages && messages.messages && messages.messages.length > 0) {
     for (const message of messages.messages) {
       if (message && !message.junk) {
-        hasNotJunk = true;
-        break;
+        if (!seenMessages[folder.accountId + folder.path].has(message.id)) {
+          seenMessages[folder.accountId + folder.path].add(message.id);
+          await window.scrNoti.notifyNativeScript(message, "new");
```

Comment:
I want the "simple" script to stay untouched, so we do have to return after the first new message for that mode..

Reply:
Ok, if the "simple" script should only receive one message, this can be done.

Done.